sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/BigQueryServicesImpl.java

```
@@ -437,67 +437,77 @@ public Table getTable(TableReference tableRef) throws InterruptedException, IOEx
         (int) TimeUnit.MINUTES.toMillis(5);
 
     /**
+     * {@inheritDoc}
+     *
      * Tries to create the BigQuery table.
      * If a table with the same name already exists in the dataset, the table
      * creation fails.  In such a case,
```

This hasn't been implemented in the original PR: "fail the createTable() when existing table's schema is different.

I added it in the patch.

---

I'm not convinced this should be implemented - users might manually update schema in backwards-compatible ways. Let's remove from PR.

---

My general feeling was silently ignore the potential error might cause bad things in the future. I will think about it more.

---

Might be worth logging at least a warning.